tests/integration/test_utils_integration.py

```diff
@@ -451,15 +448,19 @@ def test_extract_arbitrary_unitary_two_modes_vectorized(
         )
 
         if isinstance(U, tf.Tensor):
-            final_state = tf.Session().run(
-                tf.einsum("ab,b", U, tf.constant(initial_state.reshape([-1])))
-            )
-            final_state = np.tile(final_state, bsize)
-        else:
-            final_state = U @ initial_state.reshape([-1])
+            U = U.numpy()
 
-        expected_state = eng_ref.run([p0, prog]).state.ket().reshape([-1])
-        assert np.allclose(final_state, expected_state, atol=tol, rtol=0)
+        final_state = U @ initial_state.reshape([-1])
+        expected_state = eng_ref.run([p0, prog]).state.ket()
+        
+        if isinstance(expected_state, tf.Tensor):
```

Comment:
Since this is a test, shouldn't we be asserting that `expected_state` has a particular type? :thinking:

Reply:
Isn't `expected_state` a tf.Tensor only when we use the tfbackend?

Reply:
You're right, my mistake